File: client/src/models/Watcher.js
import { Account } from "./Account.js"
import { Album } from "./Album.js"

class Watcher {
  constructor(data) {
    this.id = data.id
    this.accountId = data.accountId
    this.albumId = data.albumId
  }
}

export class WatcherProfile extends Watcher {
  constructor(data) {
    super(data)
    this.profile = new Account(data)
  }
}

export class WatcherAlbum extends Watcher {
  constructor(data) {
    super(data)
    this.album = new Album(data)
    this.watcherId = data.watcherId
  }
}

File: server/Models/Watcher.cs
namespace postit.Models;

public class Watcher
{
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public string AccountId { get; set; }
  public int AlbumId { get; set; }
}

public class WatcherProfile : Account
{
  public int WatcherId { get; set; }
  public int AlbumId { get; set; }
  public string AccountId { get; set; }
}

public class WatcherAlbum : Album
{
  public int WatcherId { get; set; }
  public string AccountId { get; set; }
}